src/libopensc/card-starcos.c

```diff
@@ -1653,17 +1908,37 @@ static int starcos_decipher(struct sc_card *card,
 		apdu.lc = crgram_len + 1;
 		apdu.datalen = crgram_len + 1;
 
+		//extended length APDU
 		if (sc_get_max_send_size(card) < crgram_len + 1) {
 			/* Starcos doesn't support chaining for PSO:DEC, so we just _hope_
 			 * that both, the reader and the card are able to send enough data.
 			 * (data is prefixed with 1 byte padding content indicator) */
 			card->max_send_size = crgram_len + 1;
 			card->reader->max_send_size = crgram_len + 1;
+			//TODO: should be set in init()? -> capabilities
 			card->caps |= SC_CARD_CAP_APDU_EXT;
 		}
-
-		r = sc_transmit_apdu(card, &apdu);
-		sc_mem_clear(sbuf, crgram_len + 1);
+		//starcos_v_3_2 specific changes
+		if(card->type == SC_CARD_TYPE_STARCOS_V3_2){
+			u8 *tempbuf = NULL;
+			tempbuf = malloc(SC_MAX_APDU_BUFFER_SIZE);
```

Comment:
missing error handling; why aren't you using `out`/`outlen` directly?

Reply:
For this command, Starcos 3.2 cards will throw an error if they don't receive a buffer with the maximal possible size. Thus, we can only use the buffer in case it has the maximal size. Therefore, a temporary max-sized buffer is provided and later cropped and copied into the output buffer.

Reply:
i see. please add the error handling then